Editing posting journal test

lomamech opened the pull request:
Adding E2E test for eding transaction in the posting journal  
- Test for updating transaction
- Test for checking the function validateTransaction
- Updating Form utils and GridUtils

Nota: 
- With the new changes on the edition of accounts, it is no longer possible to register a null or invalid account,
- I have not been able to test the dates of transactions (I will create an issue on it)

closes #1505

lomamech committed:
Adding E2E test for eding transaction in the posting journal
- Test for updating transaction
- Test for checking the function validateTransaction
- Updating Form utils and GridUtils

lomamech committed:
Sanitaze Code

lomamech committed:
Adding function getCellName
- For tracking column container by col.colDef.name

lomamech committed:
Change the way to updante an account
Remove some test not necessary after the new change of edit the accounts

jniles commented on test/end-to-end/edit_journal/edit_journal.spec.js:
```diff
@@ -0,0 +1,146 @@
+    // simulate dbl click
+    browser.actions().mouseMove(accountNumberCell).doubleClick().perform();
+    
+    accountNumberCell.element(by.css("input")).sendKeys(protractor.Key.chord(protractor.Key.CONTROL, "a"));
```
Wow, this is really cool.  I didn't know you could do that.

Protractor has a method for clearing inputs ... does this not work?  It should be:
```js
accountNumberCell.element(by.css("input")).clear();
```
[Documentation](http://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.clear)

jniles commented on test/end-to-end/edit_journal/edit_journal.spec.js:
```diff
@@ -0,0 +1,146 @@
+    accountNumberCell.element(by.css("input")).sendKeys(accountKey);
+    element.all(by.css('[title="1100 - Test Capital One"]')).click();
+
+    element.all(by.css('[class="fa fa-save"]')).click();
```
These tests will break if we change our icon font.  Could you put a `data-` tag on this element and use it as a selector?  Something like `[data-method="save"]`.

jniles commented on test/end-to-end/edit_journal/edit_journal.spec.js:
```diff
@@ -0,0 +1,146 @@
+    element.all(by.css('[class="fa fa-save"]')).click();
+
+    components.notification.hasSuccess();
+    element.all(by.css('[class="ui-grid-icon-minus-squared"]')).get(0).click();
```
That is a clever way to click the group/ungroup button :+1:

jniles commented on test/end-to-end/edit_journal/edit_journal.spec.js:
```diff
@@ -0,0 +1,146 @@
+    const debitCell = GU.getCellName(gridId, 2, 5);
+    const creditCell = GU.getCellName(gridId, 3, 6);
+    // simulate dbl click
+    browser.actions().mouseMove(debitCell).doubleClick().perform();
```
This would be easier to read if you put this in a function.  Like this:
```js
// simulates a double click
const doubleClick = (element) => browser.actions().mouseMove(element).doubleClick().perform();

// ...then later...
it('edits a transaction change value of Debit and Credit', function () {
  const debitCell = GU.getCellName(gridId, 2, 5);
  doubleClick(debitCell);
});
```

jniles commented on test/end-to-end/edit_journal/edit_journal.spec.js:
```diff
@@ -0,0 +1,146 @@
+    components.notification.hasWarn();
+    element.all(by.css('[class="fa fa-ban"]')).click();
+    element.all(by.css('[class="ui-grid-icon-minus-squared"]')).get(0).click();
+    FU.buttons.grouping();
```
:+1:

jniles commented on test/end-to-end/shared/GridUtils.js:
```diff
@@ -29,6 +29,13 @@ function getCell(gridId, row, col) {
     .element(by.repeater('(colRenderIndex, col) in colContainer.renderedColumns track by col.uid').row(col));
 }
 
+function getCellName(gridId, row, col) {
```
You can just remove the old `getCell()` method.  It's obviously for an outdated version of `ui-grid`.

lomamech replied:
it's not possible to delete old function getCell, because 35 test E2E track column container by col.uid and not by col.colDef.name, it for that i have add a new function getCellName who track by col.colDef.name

jniles replied:
Oh man.  That's weird.  I wonder why...

Thanks!  Good catch :+1:

jniles reviewed (changes requested):
@lomamech, this is excellent!

I've made some comments to clean the code up and make sure it is maintainable.

 1. For each of the editing buttons, put `data-` tags and use them for protractor tests.  You are using `class` selectors, which depend on our icon font.
 2. Make `doubleClick()` a function.
 3. Delete the old `GU.getCell()` method, and rename `GU.getCellName()` to `GU.getCell()`.

Looks great!

lomamech committed:
Review Code
-Using function doubleClick
-Using protractor clear instead ctrl+A plus backspace
-Using data-methode instead icon font

lomamech committed:
Sanitaze code
